### PR TITLE
Move gather chain after devscripts-gather step

### PR DESCRIPTION
**Pull request by jluhrsen**
gather-must-gather is not working in 4.7 baremetalds because
it cannot pull the must gather image. the devscripts-gather
step brings the must-gather image locally and sets the
MUST_GATHER_IMAGE variable to that so the gather-must-gather
step will use that instead.

Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

**Comment by jluhrsen** [comment]
the [metal-ipi-ovn jobs do not rehearse properly](https://coreos.slack.com/archives/CBN38N3MW/p1620679486306000). apparently the config is too large and things choke and the [rehearsal
doesn't work.](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/18473/rehearse-18473-pull-ci-openshift-cluster-network-operator-release-4.9-e2e-metal-ipi-ovn-ipv6-ipsec/1392002968739385344) However, if you remove a few steps from the job you can get the rehearsal to run with whatever steps
you think are important. I did that with an earlier version of this PR by removing packet-teardown and e2e-test steps.
so the job in the end doesn't have [pretty results](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/18473/rehearse-18473-pull-ci-openshift-cluster-network-operator-release-4.9-e2e-metal-ipi-ovn-ipv6-ipsec/1391978243740930048), but you can see that the devscripts-gather ran first, followed by the
gather-must-gather step. in the gather-must-gather step you can see the [image being used](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/18473/rehearse-18473-pull-ci-openshift-cluster-network-operator-release-4.9-e2e-metal-ipi-ovn-ipv6-ipsec/1391978243740930048/artifacts/e2e-metal-ipi-ovn-ipv6-ipsec/gather-must-gather/artifacts/must-gather/must-gather.log) as:

`Using must-gather plug-in image: registry.redhat.io/openshift4/ose-must-gather:latest`

Currently our 4.7 version of this job is perma-failing and [blocking a release](https://coreos.slack.com/archives/CDCP2LA9L/p1620687970238900).

**Comment by stbenjam** [comment]
Looks fine to me if you fix the ordering so gather chain happens before packet teardown, but after devscripts-gather.  In my head when I wrote https://github.com/openshift/release/pull/18414 that was already the case :facepalm:   Thanks for the fix!

> the metal-ipi-ovn jobs do not rehearse properly. apparently the config is too large and things choke and the rehearsal
doesn't work. 

We should figure out why you saw this -- I haven't seen it myself, I saw on the Slack thread something was 130kB but it's not clear from the logs or the thread what was too large (maybe I just didn't have enough coffee yet).

File: ci-operator/step-registry/baremetalds/ipi/post/baremetalds-ipi-post-chain.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 chain:
   as: baremetalds-ipi-post
   steps:
-  - chain: gather
   - ref: baremetalds-devscripts-gather
   - ref: baremetalds-packet-teardown
+  - chain: gather
```
[the remark this answers is not in the document]
Comment (jluhrsen) [reply ci-operator/step-registry/baremetalds/ipi/post/baremetalds-ipi-post-chain.yaml:5]: thanks for pointing this out. This is fixed now.

**Review by wking — APPROVED** [review]
/lgtm

**Comment by stbenjam** [comment]
> > > the metal-ipi-ovn jobs do not rehearse properly. apparently the config is too large and things choke and the rehearsal
> > > doesn't work.
> > 
> > 
> > We should figure out why you saw this -- I haven't seen it myself, I saw on the Slack thread something was 130kB but it's not clear from the logs or the thread what was too large (maybe I just didn't have enough coffee yet).
> 
> I'm not sure how to get this fixed, but when I was trying to figure out what was wrong and assuming it was a problem in my
> code, I ended up checking other PRs in this area and saw that the rehearsals were failing in those PRs the same way. I guess
> the full bit of config that gets generated from the larger steps in the metal-ipi jobs is hitting some OS limit.

Just to come full circle, the rehearsal ends up embedding all the shell scripts and things into the prow job, so it is quite large. We're about to cut our TEST_SKIPS list by more than half, and we can have a look at some other optimizations like more terse comments. That should help reduce the overall size.

**Pull request merged**